### PR TITLE
Adding unit tests for Application-mgt

### PR DESCRIPTION
**Pull request by Isanka95**
Added new test class for ApplicationManagementServiceImpl class.
Added findbugs-exclude.xml file to resource.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+    private OMElement mockSystemApplicationsConfig;
+    @Mock
+    private OMElement mockChildApplicationConfig;
+    @Mock
+    private RealmService mockRealmService;
+    @Mock
+    private org.wso2.carbon.user.core.tenant.TenantManager mockTenantManager;
+
+    private static final String USERNAME = "user";
+    private static final String TENANT_DOMAIN = "carbon.super";
```
[the remark this answers is not in the document]
Comment (Isanka95) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java]: Ok

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql [hunk 2]
```diff
@@ -0,0 +1,1191 @@
...
+	   	TENANT_ID INTEGER NOT NULL, 
+	   	IDP_CLAIM VARCHAR (512) NOT NULL , 
+	   	SP_CLAIM VARCHAR (512) NOT NULL , 
+	   	APP_ID INTEGER NOT NULL, 
+	   	IS_REQUESTED VARCHAR(128) DEFAULT '0', 
+		IS_MANDATORY VARCHAR(128) DEFAULT '0', 
+	   	DEFAULT_VALUE VARCHAR(255), 
+	   	PRIMARY KEY (ID));
+
+ALTER TABLE SP_CLAIM_MAPPING ADD CONSTRAINT CLAIMID_APPID_CONSTRAINT FOREIGN KEY (APP_ID) REFERENCES SP_APP (ID) ON DELETE CASCADE;
```
Comment (ChamathNS) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql:437]: Line exceeds the 120 character limit

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql [hunk 3]
```diff
@@ -0,0 +1,1191 @@
...
+ALTER TABLE SP_ROLE_MAPPING ADD CONSTRAINT ROLEID_APPID_CONSTRAINT FOREIGN KEY (APP_ID) REFERENCES SP_APP (ID) ON DELETE CASCADE;
+
+CREATE TABLE IF NOT EXISTS SP_REQ_PATH_AUTHENTICATOR (
+	    	ID INTEGER NOT NULL AUTO_INCREMENT,
+	    	TENANT_ID INTEGER NOT NULL,
+	    	AUTHENTICATOR_NAME VARCHAR (255) NOT NULL ,
+	    	APP_ID INTEGER NOT NULL,
+	    	PRIMARY KEY (ID));
+
+ALTER TABLE SP_REQ_PATH_AUTHENTICATOR ADD CONSTRAINT REQ_AUTH_APPID_CONSTRAINT FOREIGN KEY (APP_ID) REFERENCES SP_APP (ID) ON DELETE CASCADE;
```
Comment (ChamathNS) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql:456]: Line exceeds the 120 character limit. Please fix other places as well.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+    }
+
+    private static void setInternalState(Object target, String field, Object value) {
+
+        Class c = target.getClass();
+
+        try {
+            Field f = c.getDeclaredField(field);
+            f.setAccessible(true);
+            f.set(target, value);
```
Comment (ChamathNS) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java:530]: Let's use descriptive variable names

Comment (Isanka95) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java]: Fixed

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql (hunk 3 above)
Comment (Isanka95) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql]: Fixed

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql (hunk 2 above)
Comment (Isanka95) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/resources/dbscripts/identity.sql]: Fixed with [48da372](https://github.com/wso2/carbon-identity-framework/pull/3494/commits/48da3722b0215cc4eb8f430b71d21a89731805ba)

**Review by GANGANI — APPROVED** [review]

**Pull request merged**